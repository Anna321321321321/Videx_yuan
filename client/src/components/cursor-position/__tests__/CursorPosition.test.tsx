import { mount } from 'enzyme';
import React, { Component } from 'react';
import CursorPosition from '..';

class Dummy extends Component<any, any> {
  render() {
    return <div>Dummy</div>;
  }
}

describe('CursorPosition Unit Tests', () => {
  it('Set className & style', () => {
    const wrapper = mount(
      <CursorPosition
        className={'test'}
        style={{
          height: '100px',
          width: '100px'
        }}
      />
    );
    expect(wrapper.props().className).toBe('test');
    expect(wrapper.props().style).toEqual({ height: '100px', width: '100px' });

    const cursorPositionDiv = wrapper.find('div');
    expect(cursorPositionDiv.length).toBe(1);
    expect(cursorPositionDiv.hasClass('test')).toBe(true);

    expect(wrapper).toMatchSnapshot();
  });

  it('Render child components', () => {
    const wrapper = mount(
      <CursorPosition
        style={{
          height: '100px',
          width: '100px'
        }}
      >
        <Dummy />
        <Dummy />
      </CursorPosition>
    );
    expect(wrapper.find('Dummy').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });

  it('Check child component props', () => {
    const wrapper = mount(
      <CursorPosition
        style={{
          height: '100px',
          width: '100px'
        }}
      >
        <Dummy />
      </CursorPosition>
    );
    expect(wrapper.find('Dummy').length).toBe(1);
    expect(wrapper.find('Dummy').get(0).props).toEqual({
      detectedEnvironment: {
        isMouseDetected: false,
        isTouchDetected: false
      },
      elementDimensions: {
        width: 0,
        height: 0
      },
      isActive: false,
      isPositionOutside: true,
      position: {
        x: 0,
        y: 0
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
