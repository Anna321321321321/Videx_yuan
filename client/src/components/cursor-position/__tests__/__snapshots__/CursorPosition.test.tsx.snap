// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CursorPosition Unit Tests Check child component props 1`] = `
<CursorPosition
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <ReactCursorPosition
    activationInteractionMouse="hover"
    activationInteractionTouch="press"
    hoverDelayInMs={0}
    hoverOffDelayInMs={0}
    isEnabled={true}
    mapChildProps={[Function]}
    onActivationChanged={[Function]}
    onDetectedEnvironmentChanged={[Function]}
    onPositionChanged={[Function]}
    pressDurationInMs={500}
    pressMoveThreshold={5}
    shouldDecorateChildren={true}
    shouldStopTouchMovePropagation={false}
    style={
      Object {
        "height": "100px",
        "width": "100px",
      }
    }
    tapDurationInMs={180}
    tapMoveThreshold={5}
  >
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "height": "100px",
          "width": "100px",
        }
      }
    >
      <Dummy
        detectedEnvironment={
          Object {
            "isMouseDetected": false,
            "isTouchDetected": false,
          }
        }
        elementDimensions={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        isActive={false}
        isPositionOutside={true}
        key=".0"
        position={
          Object {
            "x": 0,
            "y": 0,
          }
        }
      >
        <div>
          Dummy
        </div>
      </Dummy>
    </div>
  </ReactCursorPosition>
</CursorPosition>
`;

exports[`CursorPosition Unit Tests Render child components 1`] = `
<CursorPosition
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <ReactCursorPosition
    activationInteractionMouse="hover"
    activationInteractionTouch="press"
    hoverDelayInMs={0}
    hoverOffDelayInMs={0}
    isEnabled={true}
    mapChildProps={[Function]}
    onActivationChanged={[Function]}
    onDetectedEnvironmentChanged={[Function]}
    onPositionChanged={[Function]}
    pressDurationInMs={500}
    pressMoveThreshold={5}
    shouldDecorateChildren={true}
    shouldStopTouchMovePropagation={false}
    style={
      Object {
        "height": "100px",
        "width": "100px",
      }
    }
    tapDurationInMs={180}
    tapMoveThreshold={5}
  >
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "height": "100px",
          "width": "100px",
        }
      }
    >
      <Dummy
        detectedEnvironment={
          Object {
            "isMouseDetected": false,
            "isTouchDetected": false,
          }
        }
        elementDimensions={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        isActive={false}
        isPositionOutside={true}
        key=".0"
        position={
          Object {
            "x": 0,
            "y": 0,
          }
        }
      >
        <div>
          Dummy
        </div>
      </Dummy>
      <Dummy
        detectedEnvironment={
          Object {
            "isMouseDetected": false,
            "isTouchDetected": false,
          }
        }
        elementDimensions={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        isActive={false}
        isPositionOutside={true}
        key=".1"
        position={
          Object {
            "x": 0,
            "y": 0,
          }
        }
      >
        <div>
          Dummy
        </div>
      </Dummy>
    </div>
  </ReactCursorPosition>
</CursorPosition>
`;

exports[`CursorPosition Unit Tests Set className & style 1`] = `
<CursorPosition
  className="test"
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <ReactCursorPosition
    activationInteractionMouse="hover"
    activationInteractionTouch="press"
    className="test"
    hoverDelayInMs={0}
    hoverOffDelayInMs={0}
    isEnabled={true}
    mapChildProps={[Function]}
    onActivationChanged={[Function]}
    onDetectedEnvironmentChanged={[Function]}
    onPositionChanged={[Function]}
    pressDurationInMs={500}
    pressMoveThreshold={5}
    shouldDecorateChildren={true}
    shouldStopTouchMovePropagation={false}
    style={
      Object {
        "height": "100px",
        "width": "100px",
      }
    }
    tapDurationInMs={180}
    tapMoveThreshold={5}
  >
    <div
      className="test"
      style={
        Object {
          "WebkitUserSelect": "none",
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </ReactCursorPosition>
</CursorPosition>
`;
